packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -161,21 +161,67 @@ Follows the [Column Type Philosophy](/packages/nimble-components/src/table/specs
+
+-   `nimble-unit-byte` - Unit labels in this family are `byte`/`bytes`, `KB`, `MB`, `GB`, `TB`, `PB`
+    -   `binary` - boolean attribute that indicates a binary conversion factor of 1024 should be used rather than 1000. The resulting unit labels are `byte`/`bytes`, `KiB`, `MiB`, `GiB`, `TiB`, `PiB`.
+-   `nimble-unit-volt` - Unit labels in this family are `volt`/`volts`, plus `V` prefixed by all supported metric prefixes.
+
+Supported metric prefixes are f (femto), p (pico), n (nano), μ (micro), m (milli), c (centi), d (deci), k (kilo), M (mega), G (giga), T (tera), P (peta), and E (exa). This set is intended to be suitable for other units we may support in the future (e.g. ohms, amps), but any particular unit family can diverge from this set as needed.
+
+If a value with a unit would be formatted with exponential notation, it will always be given in its base unit. E.g. instead of "1e6 PB", it would render as "1e21 bytes".
+
+When displaying units, `Intl.NumberFormat` will translate unit strings for the [units that it supports](https://tc39.es/ecma402/#table-sanctioned-single-unit-identifiers) and localize the number (for comma/decimal). We will include our own logic for converting between unit values. For a family of units not supported by `Intl.NumberFormat`, we will provide our own translations (for French, German, Japanese, and Chinese) in a `nimble-unit-<name>` element. If the client requests a translation for one of these units in a language we don't support, we will fall back to English.
+
+Unit elements will be capable of enumerating the individual units supported. The enumerated unit objects will be capable of formatting a given number into a localized string including the unit label. Below is an example of the abstractions/APIs that might be used to implement this.
+
+```TS
+// a specific unit, e.g. kilobyte, millivolt, etc.
+interface Unit {
+    public conversionFactor: number;
+    public format(value: number): string;
+}
+
+// a set of related units, e.g. {byte, kilobyte, megabyte, gigabyte, terabyte, petabyte}
+interface UnitFamily {
+    public getSupportedUnits(lang: string, formatterOptions: Intl.NumberFormatOptions): Unit[];
+}
+
+// unit family supported by Intl.NumberFormat
+class UnitFamilyByte extends FoundationElement implements UnitFamily {
+    public getSupportedUnits(lang: string, formatterOptions: Intl.NumberFormatOptions): Unit[] {
```

Comment:
We have the NumberFormatter pattern. I think these elements should just return a NumberFormatter class that can be constructed instead of coming up with a new pattern. Maybe there would be a shared subsclass so there is a consistent interface

```ts
abstract class UnitScaleFormatter extends NumberFormatter {
    constructor (lang, options) {}
    format(number): string
}

class ByteScaleFormatter extends UnitScaleFormatter {}
```

And then the element class name can align more closely with the element name and an abstract class to define the middle interface:
```ts
class Unit extends FoundationElement {
    abstract getFormmater()
}

class UnitByte extends Unit{
    getFormmater() {return ByteScaleFormatter}
}

// somewhere in code:
const FormatterClass = el.getFormatter();
const formatter = new FormatterClass(lang, opts);
```

Reply:
One difficulty I anticipate is that a `UnitScaleFormatter` is not independent of the existing `NumberFormatter`s we have today. We have to handle formatting both decimal or default modes, with or without units. In my prototype, I accomplish this by composing `NumberFormatter`s: `DefaultFormatter` and `DecimalFormatter` internally create a `UnitScaleFormatter` to use. I'll try to see if your suggestion can work, though. The line before the code section already says that it is only "an exmple of the abstractions/APIs that might be used," so it's understood that it is subject to change.

Reply:
A problem I ran into was the return type of `Unit.getFormatter()`. My best guess is that it should be this:
```TS
abstract class Unit extends FoundationElement {
    public abstract getFormatter(): typeof UnitScaleFormatter;
}
```
But because `UnitScaleFormatter` is abstract, I get the TS error "Cannot create an instance of an abstract class" anywhere I do this:
```TS
const FormatterClass = el.getFormatter();
const formatter = new FormatterClass(lang, opts);
```

Reply:
You use the `new` keyword for typing constructors. Can see examples in the mixins. [playground example](https://www.typescriptlang.org/play?#code/FASwdgLgpgTgZgQwMZQAQFUwggZSQgGygDEB7GAWwQmhgHkAHCEUsAZ1QG9hVfUA3QgC5UbCDHABzANzAAvsAQAjMTGQRUSAgjYdM2PIRLkqNWFx58krVQFckEcgApSTFuxH7c+ImUrVaRmYbAEouBT5UZVV1VDgTaicQkVUpWQUtHQ4AIQBPaD9TWlQoAA9oMAATPSxvI0KA825I+P8IJJTxKQtIyJgoCFsYMFQAcgALKAICUlHZSIUFCFyGNC9DXwSzGABhVnF7RxhUAF5UMCgAd1QXNxtPWo3jNsC79jCTgD4MR59notgsms7A0kgGDW2pxuyR+Bj+ENoe0gMEO5FO32afH6g2GqDyBS2tHSwGBYlQCPMZzBEApMCSQJsGlaAOOZwu11pTk4ggIIgmIFGchCDPYpCIADoZpInMzGjBxbL2iFhcAgA)

Reply:
Not sure how the response meant this was resolved so re-opening.
The response didn't sound like a resolution but continued discussion... I'm not sure what the intention of resolving it was.

Reply:
Yeah, based on the short chat we had about it last week, my take was that we didn't need to settle on implementation details in this thread/PR, as long as it was understood that the API I had in the spec was not the decided upon approach -- just an example. I am still making an effort to use your suggested approach in my prototype.

Reply:
Gotcha, my concerns where:
- the comment sounded like something was tried and didn't work but didn't say conversation would continue anywhere else
- my follow-up response with an example was not acknowledged, the thread was just resolved without discussion

Would be helpful to make sure it's clear how a thread is resolved (thumbs up or comment, etc).